### PR TITLE
Default offline store

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
This PR makes the offline store the default redux store.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [] All new/modified APIs include changes to the drivers

#### Test Information
This PR was tested on:
iPhone 6 iOS 10
iPhone 7 iOS 10
Galaxy s7 - Android 6.0 API 23

@jarredwitt

### CODE REVIEW

**Comment by enahum** [comment]
@jarredwitt can you check why is this failing to build?

**Comment by jarredwitt** [comment]
@enahum good to go.

File: src/reducers/entities/preferences.js [hunk 1]
```diff
@@ -10,20 +10,26 @@ function getKey(preference) {
 
 function myPreferences(state = {}, action) {
     switch (action.type) {
+    case PreferenceTypes.SAVE_PREFERENCES:
```
Comment (enahum) [inline src/reducers/entities/preferences.js]: @jarredwitt can we remove this as it doesn't even exist

File: src/reducers/entities/preferences.js [hunk 2]
```diff
@@ -10,20 +10,26 @@ function getKey(preference) {
...
-            nextState[getKey(preference)] = preference;
+        if (action.data) {
+            for (const preference of action.data) {
+                nextState[getKey(preference)] = preference;
+            }
         }
 
         return nextState;
     }
+    case PreferenceTypes.DELETE_PREFERENCES:
```
Comment (enahum) [inline src/reducers/entities/preferences.js]: what about this one?

**Comment by jarredwitt** [comment]
@enahum changes made.

File: package.json [hunk 3]
```diff
@@ -29,9 +29,12 @@
     "eslint-plugin-mocha": "4.9.0",
     "fetch-mock": "5.9.4",
     "form-data": "2.1.2",
+    "jsdom": "^9.12.0",
+    "jsdom-global": "^2.1.1",
     "mocha": "3.2.0",
     "nock": "9.0.11",
-    "react": "15.4.2",
+    "react": "^15.4.2",
+    "redux-persist-node-storage": "^1.0.2",
```
Comment (jwilander) [inline package.json]: Can we remove the `^` so we don't auto-upgrade?

File: src/actions/users.js [hunk 4]
```diff
@@ -643,22 +643,26 @@ export function stopPeriodicStatusUpdates() {
...
+        const state = getState();
+        const currentUserId = state.entities.users.currentUserId;
+        const currentNotifyProps = state.entities.users.profiles[currentUserId].notify_props;
+
+        dispatch({
+            type: UserTypes.UPDATE_NOTIFY_PROPS,
+            notifyProps,
+            meta: {
+                offline: {
+                    effect: () => Client.updateUserNotifyProps(notifyProps),
```
Comment (jwilander) [inline src/actions/users.js]: Can we use `Client4.patchMe({notify_props: notifyProps});` here like it was before?

**Comment by jarredwitt** [comment]
@jwilander changes made.

File: package.json [hunk 5]
```diff
@@ -29,9 +29,12 @@
     "eslint-plugin-mocha": "4.9.0",
     "fetch-mock": "5.9.4",
     "form-data": "2.1.2",
+    "jsdom": "^9.12.0",
+    "jsdom-global": "^2.1.1",
     "mocha": "3.2.0",
     "nock": "9.0.11",
     "react": "15.4.2",
+    "redux-persist-node-storage": "^1.0.2",
```
Comment (jwilander) [inline package.json]: Sorry, can we remove the `^` from all these new packages too?

**Comment by jarredwitt** [comment]
@jwilander ready for you again.

**Comment by jwilander** [comment]
Awesome, good stuff. Excited to start using this

**Pull request merged**